src/storage/page_buffer.c

```diff
@@ -14958,6 +14988,25 @@ pgbuf_bcb_is_dirty (const PGBUF_BCB * bcb)
   return (bcb->flags & PGBUF_BCB_DIRTY_FLAG) != 0;
 }
 
+static bool
+pgbuf_bcb_is_dirty_and_needs_flushing (PGBUF_BCB * bcb)
+{
+  if (!pgbuf_bcb_is_dirty (bcb))
+    {
+      // is not dirty
+      return false;
+    }
+
+  if (bcb->flags & PGBUF_BCB_FLUSH_NOT_NEEDED)
```

Comment:
pedantic:
```suggestion
  if ((bcb->flags & PGBUF_BCB_FLUSH_NOT_NEEDED) == PGBUF_BCB_FLUSH_NOT_NEEDED)
```

Reply:
What would this change be for? Readability?

Reply:
It suggests to the reader that `PGBUF_BCB_FLUSH_NOT_NEEDED` is a "proper" (single) flag. Whereas, for example, `FLAG_BCB_IS_BEING_VICTIMIZED` defined elsewhere is not a proper flag.
By the way, `PGBUF_BCB_FLUSH_NOT_NEEDED` seems to break a pseudo-naming-convention as other flags end in `_FLAG`.

Reply:
The `_FLAG` suffix was added to the name, it should suggest the reader it is a "proper" single flag. As for "is flag set" check, I prefer the shorter version with the same result (as long as the shorter version is readable).

Reply:
Fair. As said in the beginnig the suggestion is "pedantic" :) .